pages/creating_unifex_natives.md

```diff
@@ -0,0 +1,167 @@
+        interface: [:nif, :cnode]
+      ],
+   ```
+   but at this moment we have to do it in a little more descriptive way.
+   Source files have to be located in `c_src/example` directory.
+   More details on how to use bundlex can be found in its [documentation](https://hexdocs.pm/bundlex).
+
+## Native code
+
+Let's start by creating a `c_src/example` directory, and the files that will be needed:
+
+```bash
+mkdir -p c_src/example
+cd c_src/example
+touch example.c
+touch example.h
+touch example.spec.exs
+```
+
+You may wonder where is the `_generated/nif/example.c` and `_generated/cnode/example.c`. 
+Well, as the name suggests, it will be generated based on `example.spec.exs`!
+
+Here are the contents of `example.spec.exs`:
+
+```elixir
+module Example
+
+interface [NIF, CNode]
+
+spec foo() :: {:ok :: label, answer :: int}
+```
+
+Note that here we also specified an interface or even interfaces!
+It is not necessary because if we didn't do it Unifex would take it from `bundlex.exs`.
+However, this is a good practice that makes code clearer and is also a little faster than fetching info from `bundlex.exs.` 
+
+More information on how `.spec.exs` files should be created can be found in docs for
+`Unifex.Specs.DSL` module.
+You can also check out our [test_projects](https://github.com/membraneframework/unifex/tree/master/test_projects)
+to see a little more advanced examples or look how we use Unifex in a our repositories 
+e.g. in [shmex](https://github.com/membraneframework).
+
+Next step is to implement our `example.h`.
+Since our example is very simple our `example.h` will be very simple too:
+```c
+#include "_generated/example.h"
+```
+
+It just includes generated header file that contains some function definitions we use in our `example.c`. 
+
+Now, let's provide required implementation in `example.c`:
+
+```c
+#include "example.h"
+
+UNIFEX_TERM foo(UnifexEnv* env) {
+  return foo_result_ok(env, 10);
+}
+```
+It is a very simple C code that always returns 10.
+
+At this moment the project should successfully compile. 
+Run `mix deps.get && mix compile` to make sure everything is fine.
+In `c_src/_generated/` directory there should appear files both for usage as NIF and CNode.
+
+## Running code
+
+### NIF
+
+All you have to do in order to access natively implemented functions is to create a module with the name as defined 
+in `example.spec.exs` and to use `Unifex.Loader` there:
+
+```elixir
+defmodule Example do
+  use Unifex.Loader
+end
+```
+
+And that's it! You can now run `iex -S mix` and check it out yourself:
+
+```elixir
+iex(1)> Example.foo()
+{:ok, 10}
+```
+
+### CNode
+In case of `CNodes`, module with `Unifex.Loader` is unnecessary. We would just do:
+```elixir
+iex(2)> require Unifex.CNode
+Unifex.CNode
+iex(3)> {:ok, cnode} = Unifex.CNode.start_link(:example)
+
+10:56:38.259 [info]  Protocol 'inet_tcp': register/listen error: econnrefused
+
+ 
+10:56:38.259 [info]  Trying to start epmd...
```

Comment:
Should this be returned?

Reply:
The first message is logged by Erlang stdlib when it can't connect to CNode because empd is not started, so there's no way to disable it. The other is logged by Bundlex. It won't happen if epmd is started, so you can remove that from here. See https://erlang.org/doc/man/epmd.html